test/kubernetes_unittest.cc

```diff
@@ -1863,15 +1863,14 @@ TEST_F(KubernetesTestFakeServerOneWatchRetryNodeLevelMetadata,
 
   MetadataStore store(*config);
   KubernetesUpdater updater(*config, /*health_checker=*/nullptr, &store);
-  std::thread updater_thread([&updater] { updater.Start(); });
```

Comment:
Contrary to the header comment, updaters are not guaranteed to be asynchronous. For all you know, `Start()` may be a polling loop. That said, we do know that a `KubernetesUpdater` *is* asynchronous, so this specific change is fine.